File: Assembly-FSharp/Assembly-FSharp/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/AnimatorModel.fs
namespace Styx.Models

open UnityEngine

[<CreateAssetMenu(fileName = "Animator Model", menuName = "Models/Animator model")>]
type AnimatorModel () =
    inherit ScriptableObject ()

    [<SerializeField>]
    let mutable animator: Animator = null

    member this.Animator
        with set(v) = animator <- v

    member this.SetTrigger (trigger: string) =
        animator.SetTrigger(trigger)

    member this.SetFloat (key: string) (value: float32) =
        animator.SetFloat (key, value)

File: Assets/_Scripts/Models/AnimatorModel.cs
using UnityEngine;

namespace Styx.Models
{
    [CreateAssetMenu(fileName = "Animator Model", menuName = "Models/Animator model")]
    public class AnimatorModel : ScriptableObject
    {
        [SerializeField]
        private Animator _animator;

        public Animator Animator
        {
            get { return _animator; }
            set { _animator = value; }
        }

        public void SetTrigger (string trigger)
        {
            _animator.SetTrigger(trigger);
        }

        public void SetFloat (string key, float value)
        {
            _animator.SetFloat(key, value);
        }

        public void Disable()
        {
            if (_animator.enabled)
                _animator.enabled = false;
        }

        public void Enable()
        {
            if (!_animator.enabled)
                _animator.enabled = true;
        }
    }
}
